metafields.go

```diff
@@ -0,0 +1,196 @@
+package shopify
+
+import (
+	"strconv"
+	"time"
+)
+
+// Metafields are a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafields []Metafield
+
+// GetByKey gets a metafield via its key
+func (m Metafields) GetByKey(key string) (interface{}, error) {
+	var err error
```

Comment:
Do you need to declare this err?

Reply:
Might not need to yet. Im going to write a test for this method also I think to know it works for all types. It's actually missing some!